tests/functional/build/test_build.py

```diff
@@ -0,0 +1,192 @@
+    def models(self, models_circular_relationship):  # noqa: F811
+        return {
+            "test.yml": models_circular_relationship["test.yml"],
+            "model_0.sql": models_circular_relationship["model_0.sql"],
+            "model_1.sql": models_circular_relationship["model_1.sql"],
+            "model_99.sql": models_circular_relationship["model_99.sql"],
+        }
+
+    def test__circular_relationship_test_success(
+        self,
+        project,
+    ):
+        """Ensure that tests that refer to each other's model don't create
+        a circular dependency."""
+        results = self.build()
+        actual = [r.status for r in results]
+        expected = ["success"] * 7 + ["pass"] * 2
+        assert sorted(actual) == sorted(expected)
+
+
+class TestSimpleBlockingTest(TestBuildBase):
+    @pytest.fixture(scope="class")
+    def models(self, models_simple_blocking):  # noqa: F811
+        return {
+            "schema.yml": models_simple_blocking["schema.yml"],
+            "model_b.sql": models_simple_blocking["model_b.sql"],
+            "model_a.sql": models_simple_blocking["model_a.sql"],
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "snapshot-paths": ["does-not-exist"],
+            "seed-paths": ["does-not-exist"],
+        }
+
+    def test__simple_blocking_test(
+        self,
+        project,
+    ):
+        """Ensure that a failed test on model_a always blocks model_b"""
+        results = self.build(expect_pass=False)
+        actual = [r.status for r in results]
+        expected = ["success", "fail", "skipped"]
+        assert sorted(actual) == sorted(expected)
+
+
+class TestInterdependentModels(TestBuildBase):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "snapshot-paths": ["snapshots-none"],
+            "seeds": {
+                "quote_columns": False,
+            },
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self, models_interdependent):  # noqa: F811
+        return {
+            "schema.yml": models_interdependent["schema.yml"],
+            "model_c.sql": models_interdependent["model_c.sql"],
+            "model_a.sql": models_interdependent["model_a.sql"],
+        }
+
+    def tearDown(self):
+        if os.path.exists(normalize("models-interdependent/model_b.sql")):
+            os.remove(normalize("models-interdependent/model_b.sql"))
+
+    def test__interdependent_models(self, project, test_data_dir):
+        # check that basic build works
+        copy_file(test_data_dir, "model_b.sql", project.project_root + "/models", ["model_b.sql"])
+        results = self.build()
+        assert len(results) == 16
+
+        # return null from model_b
+        copy_file(
+            test_data_dir, "model_b_null.sql", project.project_root + "/models", ["model_b.sql"]
+        )
```

Comment:
Why does this file and the `model_b.sql` file exist in the `/data` and in `fixtures`?

Reply:
Because I forgot to remove them from fixtures :)
updated